spec/models/conversion_host_spec.rb

```diff
@@ -5,16 +5,17 @@
 
   context "provider independent methods" do
     let(:host) { FactoryGirl.create(:host) }
-    let(:vm) { FactoryGirl.create(:vm_or_template) }
-    let(:conversion_host_1) { FactoryGirl.create(:conversion_host, :resource => host) }
-    let(:conversion_host_2) { FactoryGirl.create(:conversion_host, :resource => vm) }
+    let(:vm) { FactoryGirl.create(:vm) }
+    let(:conversion_host_1) { FactoryGirl.create(:conversion_host, :resource => host, :resource_type => 'Host', :address => '10.0.0.1') }
+    let(:conversion_host_2) { FactoryGirl.create(:conversion_host, :resource => vm, :resource_type => 'Vm', :address => '10.0.1.1') }
     let(:task_1) { FactoryGirl.create(:service_template_transformation_plan_task, :state => 'active', :conversion_host => conversion_host_1) }
     let(:task_2) { FactoryGirl.create(:service_template_transformation_plan_task, :conversion_host => conversion_host_1) }
     let(:task_3) { FactoryGirl.create(:service_template_transformation_plan_task, :state => 'active', :conversion_host => conversion_host_2) }
 
     before do
       allow(conversion_host_1).to receive(:active_tasks).and_return([task_1])
       allow(conversion_host_2).to receive(:active_tasks).and_return([task_3])
+      allow(conversion_host_2).to receive(:resource).and_return(vm)
```

Comment:
?? shouldn't setting `:resource` on the factory do this automatically

Reply:
@agrare This was the source of some considerable consternation and I had to pull in @gmcculloug to help me resolve it. From the v2v chat:

```
@djberg96 the issue is that the VM object is being loaded somewhere so the the instance of
the object that you call allow(vm).to receive(:ipaddresses).and_return... on is not the exact
same object (they have different object_id values) as the one setup in the test.
```

Reply:
So the spec passes without this mock here for me locally, can you delete it?  It really shouldn't be necessary.

Reply:
@agrare ok, done.